### PR TITLE
Allow to replace hibernate current session class

### PR DESCRIPTION
**Pull request by dstepanov**
Followup https://github.com/micronaut-projects/micronaut-sql/issues/372
This should remove the need to have `DataEntityManagerFactoryBean`.

### CODE REVIEW

File: hibernate-jpa/src/main/java/io/micronaut/configuration/hibernate/jpa/EntityManagerFactoryBean.java
Comment (graemerocher) [inline hibernate-jpa/src/main/java/io/micronaut/configuration/hibernate/jpa/EntityManagerFactoryBean.java:109]: This may need to use `containsBean` with a fallback for the case where it is missing?

**Review by graemerocher — CHANGES_REQUESTED** [review]

File: hibernate-jpa/src/main/java/io/micronaut/configuration/hibernate/jpa/EntityManagerFactoryBean.java
Comment (dstepanov) [reply hibernate-jpa/src/main/java/io/micronaut/configuration/hibernate/jpa/EntityManagerFactoryBean.java:109]: There is a default `MicronautHibernateCurrentSessionContextClassProvider` or I can remove it and return `MicronautSessionContext` if no provider exists.

**Comment by dstepanov** [comment]
Any idea why does it fail with `Error: Resource not accessible by integration`?

File: hibernate-jpa/src/main/java/io/micronaut/configuration/hibernate/jpa/EntityManagerFactoryBean.java
Comment (dstepanov) [reply hibernate-jpa/src/main/java/io/micronaut/configuration/hibernate/jpa/EntityManagerFactoryBean.java]: Done

**Comment by dstepanov** [comment]
@graemerocher Can this be merged if it's fine? Compile-time proxies don't work with data at the moment 😭 
I would create a PR for the data project and hopefully, it can make it to 2.2.1

**Comment by dstepanov** [comment]
@graemerocher Is there a minor release planned? It would be nice to have it then fix the data so it can make into 2.2.1

**Comment by ilopmar** [comment]
@dstepanov if you want a new release I can do it

**Comment by dstepanov** [comment]
@ilopmar Please do, thanks!

**Comment by graemerocher** [comment]
@ilopmar I just merged https://github.com/micronaut-projects/micronaut-sql/pull/356 which includes a Hibernate upgrade we should include since there was a recently announced CVE in the Hibernate release we ship with

**Comment by graemerocher** [comment]
See https://in.relation.to/2020/11/19/hibernate-orm-5424-final-release/

**Comment by ilopmar** [comment]
I'll do the release today.

**Comment by ilopmar** [comment]
The release is out and I'm waiting for CI to finish before merge [the bom upgrade](https://github.com/micronaut-projects/micronaut-core/pull/4602)

**Pull request merged**
